src/lavinmq/http/controller/queues.cr

```diff
@@ -41,6 +42,26 @@ module LavinMQ
           end
         end
 
+        get "/api/queues/:vhost/:name/unacked" do |context, params|
+          with_vhost(context, params) do |vhost|
+            refuse_unless_management(context, user(context), vhost)
+            unacked_messages = Array(UnackedMessage).new
+            @amqp_server.vhosts[vhost].queues[params["name"]].@consumers.each do |c|
+              c.unacked_messages.each do |u|
```

Comment:
This will remove the array:
```suggestion
            consumers = @amqp_server.vhosts[vhost].queues[params["name"]].@consumers.each
            unacked_messages = consumers.flat_map do |c|
              c.unacked_messages.each.compact_map do |u|
```

Reply:
… and pass `unacked_messages` to `page`, not `unacked_messages.each`.

Reply:
Thanks! Will give it a try tonight or tomorrow 👍